.github/workflows/CI-tests.yml

```diff
@@ -0,0 +1,100 @@
+name: CI tests
+
+on: [push]
+
+env:
+  poetry-version: 1.1.2
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache Poetry virtual env
+      uses: actions/cache@v2
+      env:
+          cache-name: cache-poetry-virtualenv
+      with:
+          path: .venv
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-python-${{ matrix.python-version }}-${{ hashFiles('pyproject.toml') }}
+    - name: Run image
+      uses: abatilo/actions-poetry@v2.0.0
+      with:
+        poetry-version: ${{ env.poetry-version }}
+    - name: Install Python packages
+      run: |
+          poetry config virtualenvs.in-project true
+          poetry install
+
+  lint:
```

Comment:
suggestion: use [pre-commit](https://pre-commit.com/) and the relevant GitHub action

e.g. https://github.com/torchbox/wagtailmedia/blob/main/.github/workflows/test.yml#L14-L22 / https://github.com/torchbox/wagtailmedia/blob/main/.pre-commit-config.yaml